### ISSUE TITLE
Examples for number[] does not validate with `resourceTypes` defined

### ISSUE THREAD

**Issue by rocketraman** [raml-org/raml-js-parser-2#272]
Similar to #134 , a `number[]` with `0` does not validate (using api workbench 0.8.31 and parser 0.2.24):

Does not validate:

```
  GeoPoint:
    properties:
      coordinates:
        type: number[]
    example:
      coordinates: [100.0, 0.0]
```

but does validate:

```
  GeoPoint:
    properties:
      coordinates:
        type: number[]
    example:
      coordinates: [100.0, 1.0]
```

**Comment by KonstantinSviridov** [comment]
Hi, @rocketraman 

I've just tried your example with the latest published parser version, but it worked fine for me.

Could you, please, confirm, that the issue is reproduced with the latest publishe version (0.2.24)?

**Comment by rocketraman** [comment]
Seems to be:

```
$ apm list
...
/home/raman/.atom/packages (10)
├── api-workbench@0.8.31
...
```

and:

https://github.com/mulesoft/api-workbench/blob/v0.8.31/package.json#L61

Here is a screenshot of the problem:

![askpal1](https://cloud.githubusercontent.com/assets/53049/16080411/b64f131e-32d6-11e6-9e80-edd932906883.png)

**Comment by rocketraman** [comment]
Here is a repro:

https://gist.github.com/rocketraman/1a86bedb4356011444cdd5f512be1f55

Oddly, it seems to be the presence of the `resourceTypes` section that is associated with the validation problem. The errors disappear if `resourceTypes` is removed.

**Comment by sichvoge** [comment]
Just wanted to post that if you isolate the type it works. :)

![image](https://cloud.githubusercontent.com/assets/498114/16080634/b014b14a-3301-11e6-9a72-ff83d9f315fd.png)

**Comment by rocketraman** [comment]
Ok got the repro down to this:

![image](https://cloud.githubusercontent.com/assets/53049/16080801/84c1936a-32d8-11e6-81ec-5908407010b1.png)

**Comment by KonstantinSviridov** [comment]
Ok, it is reproduced when I add resource types

**Comment by rocketraman** [comment]
I see this also occurs for non-number types:

![image](https://cloud.githubusercontent.com/assets/53049/16172038/30c3fc22-354d-11e6-9bff-ae6e3272a66d.png)

 I'm assuming the fix also works for the above?

**Comment by rocketraman** [comment]
Confirming the issue for both number and non-number types as per previous comment is fixed on api-workbench 0.8.32 using parser 0.2.25. Thanks!
